blackduck-artifactory-common/src/main/java/com/synopsys/integration/blackduck/artifactory/modules/scan/ScanModuleConfig.java

```diff
@@ -81,7 +91,38 @@ public static ScanModuleConfig createFromProperties(
         Boolean codelocationIncludeHostname = configurationPropertyManager.getBooleanProperty(ScanModuleProperty.CODELOCATION_INCLUDE_HOSTNAME);
         Boolean metadataBlockEnabled = configurationPropertyManager.getBooleanProperty(ScanModuleProperty.METADATA_BLOCK);
 
-        return new ScanModuleConfig(enabled, cron, binariesDirectoryPath, artifactCutoffDate, dryRun, namePatterns, memory, repoPathCodelocation, repos, codelocationIncludeHostname, metadataBlockEnabled, dateTimeManager);
+        // Policy properties
+        Boolean policyBlockedEnabled = configurationPropertyManager.getBooleanProperty(ScanModuleProperty.POLICY_BLOCK);
+        List<String> policyRepos = configurationPropertyManager.getRepositoryKeysFromProperties(ScanModuleProperty.POLICY_REPOS, ScanModuleProperty.POLICY_REPOS_CSV_PATH)
+                                       .stream()
+                                       .filter(artifactoryPAPIService::isValidRepository)
+                                       .filter(repos::contains)
+                                       .collect(Collectors.toList());
+        if (policyRepos.isEmpty()) {
+            policyRepos = repos;
+        }
+        List<PolicyRuleSeverityType> policySeverityTypes = configurationPropertyManager.getPropertyAsList(ScanModuleProperty.POLICY_SEVERITY_TYPES)
+                                                               .stream()
+                                                               .map(PolicyRuleSeverityType::valueOf)
+                                                               .collect(Collectors.toList());
+
+        return new ScanModuleConfig(
+            enabled,
+            cron,
+            binariesDirectoryPath,
+            artifactCutoffDate,
+            dryRun,
+            namePatterns,
+            memory,
+            repoPathCodelocation,
+            repos,
+            codelocationIncludeHostname,
+            metadataBlockEnabled,
+            policyBlockedEnabled,
+            policyRepos,
+            policySeverityTypes,
+            dateTimeManager
+        );
```

Comment:
This config seems very complicated to build. Perhaps we should have a ScanModuleConfigBuilder in the future?

Reply:
I agree. I'll add a TODO and make a ticket.